src/file/file.repository.ts

```diff
@@ -32,8 +33,12 @@ export default class FileRepository {
    * @param id - the file id.
    * @param partialFile - the partial file containing the attributes to be changed.
    */
-  static updateById(id: string, partialFile: Partial<IFile>): Promise<IFile | null> {
-    return fileModel.findByIdAndUpdate(id, partialFile, { new: true, runValidators: true }).exec();
+  static async updateById(id: string, partialFile: Partial<IFile>): Promise<boolean> {		
+		const file = await fileModel.findById(id);
+		if (!file) throw new FileNotFoundError();
+
+		const res = await file.updateOne(partialFile, { runValidators: true }).exec();
+		return res.n == 1 && res.nModified == 1 && res.ok == 1;
```

Comment:
what if res.n >1 and res.nModified ==1 ? 
would you still return false?

Reply:
https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/#update
`n` is the number of documents matched with the condition which couldn't be bigger than 1 if the condition is on a unique index such as the model's id, `nModified` is the number of documents that were modified which would always be 1 as long as `n` is >=1 and the update is done using `updateOne`. So there's no possibility that `res.n > 1` AND `res.nModified == 1`, but if it would be possible then it would obviously return `false` as `res.n == 1` would evaluate to `false`.

Reply:
accepted